shiny/render/_dataframe.py

```diff
@@ -207,6 +207,10 @@ def to_payload(self) -> object:
 
 
 def serialize_pandas_df(df: "pd.DataFrame") -> dict[str, Any]:
+    # Currently, we don't make use of the index; drop it so we don't error trying to
+    # serialize it or something
+    df = df.reset_index(drop=True)
```

Comment:
Does it matter that this could alter the data frame that the user has passed in?

Perhaps when we do Pandas data frame operations, we should enable copy on write. This can be done in blocks of code like this:

```python
with pd.option_context('mode.copy_on_write', True):
    # Do operations here
    ...
```

(It would be dangerous to set the global option `pd.options.mode.copy_on_write = True`.)

Reply:
It looks like CoW was first available in Pandas 1.5.0, which was released about a year ago.
https://pandas.pydata.org/docs/dev/user_guide/copy_on_write.html

Reply:
`reset_index` has an argument `inplace=False`. I assumed this means I'm not mutating the original?

Reply:
Here's evidence: https://github.com/pandas-dev/pandas/blob/0f437949513225922d851e9581723d82120684a6/pandas/core/frame.py#L6151-L6154

Reply:
Sounds good - I was able to confirm by running the example from [here](https://pandas.pydata.org/docs/dev/user_guide/copy_on_write.html#description) in shinylive, and it does indeed make a copy. It's pretty confusing to figure out when Pandas makes a copy and when it doesn't.

```
>>> df = pd.DataFrame({"foo": [1, 2, 3], "bar": [4, 5, 6]})
>>> df2 = df.reset_index(drop=True)
>>> df2.iloc[0, 0] = 100
>>> df
   foo  bar
0    1    4
1    2    5
2    3    6
>>> df2
   foo  bar
0  100    4
1    2    5
2    3    6
```